b2share/modules/records/utils.py

```diff
@@ -45,3 +43,14 @@ def is_deposit(record):
         bool: True if the record is a deposit record, else False.
     """
     return record.json['$schema'].endswith('#/draft_json_schema')
+
+
+
+
+def list_db_published_records():
+    """A generator for all the published records"""
+    query = RecordMetadata.query.filter(RecordMetadata.json != None)
```

Comment:
It can wait for now as it is used only in the CLI. However it doubles the number of records returned by the DB as it goes through the deposits too.